File: softuni-js-basics/06-nested-loops/exercises-submitted-evaluated/04.sumOfTwoNumbers.js
function sumOfTwoNumbers(input) {
    let start = Number(input[0]);
    let end = Number(input[1]);
    // end IS GREATER than start (rule - default)
    let magic = Number(input[2]);
    let counter = 0;
    let isFound = false;

    for (let x1 = start; x1 <= end; x1++) {
        for (let x2 = start; x2 <= end; x2++) {
            counter++;
            if (x1 + x2 === magic) {
                isFound = true;
                console.log(`Combination N:${counter} (${x1} + ${x2} = ${magic})`);
                break;
            }
        }
        if (isFound) {
            break;
        }
    }

    if (!isFound) {
        console.log(`${counter} combinations - neither equals ${magic}`);
    }

}

sumOfTwoNumbers(["1", "10", "5"]);
sumOfTwoNumbers(["23", "24", "20"]);
sumOfTwoNumbers(["88", "888", "1000"]);
sumOfTwoNumbers(["88", "888", "2000"]);

File: softuni-python-basics/06_nested_loops/04_sum_of_two_numbers.py
num1 = int(input())
num2 = int(input())
magic = int(input())
combination_counter = 0
combination_found = False

for i in range(num1, num2 + 1):
    for j in range(num1, num2 + 1):
        result = i + j
        combination_counter += 1
        if result == magic:
            print(f"Combination N:{combination_counter} ({i} + {j} = {result})")
            combination_found = True
            break
    if combination_found:
        break
if not combination_found:
    print(f"{combination_counter} combinations - neither equals {magic}")
